### ISSUE TITLE
How to extend knockout observables with Decorators using Typescript?

### ISSUE THREAD

**Issue by Eisenspalter** [knockout/knockout#2013]
I want to use decorators for knockout things like validation. Stackoverflow gives me no answer. Is there a way to do this with knockout? This would be very stylish.

[Stackoverflow](http://stackoverflow.com/questions/33858657/how-to-extend-knockout-observables-with-decorators-using-typescript)

**Comment by IanYates** [comment]
I agree, this would be awesome.  I only just started reading about decorators last week.  When I start using them in anger I'll give this some thought and post back.

**Comment by jmvtrinidad** [comment]
Hi @Eisenspalter,
Using decorator from TypeScript will only do away the process of .extend({ /*validaiton here*/}).But some of the option from required validation needs to be defined and that will render the code not desirable. Something like the following:
```
class PersonViewModel {
    @required(param: true, onlyIf: function(){ return true:/*True if needs to validate*/})
    firstname = ko.observable<string>();
}
```
What do you think?

**Comment by jmvtrinidad** [comment]
You can check this [project](https://github.com/variotry/knockout-decorator), but it has a problem, all property decorated with observable will be added as a prototype of that object. So just stay simple, stick with extend ✌
